micromouse/RobotIO.cpp

```diff
@@ -184,21 +221,74 @@ namespace Micromouse
 		leftMotor.brake();
 		rightMotor.brake();
 #endif
+*/
+
 	}
 
-	// ## NOT YET IMPLEMENTED ##
-	void RobotIO::moveForward()
-	{
 
 
+	void RobotIO::moveForward(float millimeters)
+	{		//centimeters represents how much farther the bot needs to travel.
+			//The function will loop until centimeters is within DISTANCE_TOLERANCE
+
+		leftMotor.setMaxSpeed(0.25f);
+		rightMotor.setMaxSpeed(0.25f);
+		leftMotor.resetCounts();
+		rightMotor.resetCounts();
+
+		distPID.start(millimeters);
+		headingPID.start(estimateHeadingError());
+
+		while (millimeters > DISTANCE_TOLERANCE || millimeters < -DISTANCE_TOLERANCE)
+		{
+
+			//Get distance traveled in cm since last cycle (average of two encoders)
+			float traveled = leftMotor.resetCounts() + rightMotor.resetCounts(); //resetCounts() also returns counts
```

Comment:
i hope resting so many times doesnt cause us to miss counts .. should be fine though